src/rgw/rgw_aws_lambda.cc

```diff
@@ -0,0 +1,209 @@
+//
+// Created by prateek on 30/03/20.
+//
+
+#include <string>
+#include <iostream>
+#include <utility>
+#include "common/dout.h"
+#include <aws/core/Aws.h>
+#include <aws/core/utils/Outcome.h>
+#include <aws/lambda/LambdaClient.h>
+#include <aws/lambda/model/InvokeRequest.h>
+#include <boost/lockfree/queue.hpp>
+#include <aws/core/utils/ARN.h>
+#include <aws/core/auth/AWSCredentials.h>
+#include <unordered_set>
+
+
+namespace rgw::lambda {
+  static const int STATUS_OK = 1;
+  static const int STATUS_FULL = 2;
+  static const int STATUS_MANAGER_STOPPED = 3;
+  static const int STATUS_CLIENT_CLOSED = 4;
+
+  struct client_id_t {
+    const std::string &accessKey;
```

Comment:
i think that using refs here is an issue.
you pass in refs to local variables in the `connect()` call, or the function that invoke the `connect()` call, this would keep a reference to invalid memory here (when you put that into the client map).
i think it should be just a const member